### ISSUE TITLE
SqlProgrammabilityProvider doesnt works on Mono

### ISSUE THREAD

**Issue by xkrt** [fsprojects/FSharp.Data.SqlClient#237]
It seems SqlProgrammabilityProvider doesnt works with Mono: https://twitter.com/mitekm/status/758900343358959617

### Related information 

* Linux
* ver 1.8.2
* SQL Server 2014 Developer Edition
* Mono 4.4.2
* Monodevelop 5.10

**Comment by varon** [comment]
Did you find a workaround for this? Any response from the maintainers yet?

**Comment by dmitry-a-morozov** [comment]
I have no plans to roll-out Mono support myself. 
Too much effort because of in-comparability.
But PRs are welcome.

**Comment by varon** [comment]
Thanks for the answer, so Mono isn't officially supported, then?

**Comment by dmitry-a-morozov** [comment]
I had this support before for SqlCommandProvider.
It lead to ridiculously complex code like

https://github.com/fsprojects/FSharp.Data.SqlClient/blob/f76e5068c13e87a56eebbc205f19afcc6398fb06/src/SqlClient/SqlClientExtensions.fs#L611

https://github.com/fsprojects/FSharp.Data.SqlClient/blob/f76e5068c13e87a56eebbc205f19afcc6398fb06/src/SqlClient/ISqlCommand.fs#L191

and other complications.

Until Mono sql client implementation will follow some basic quality standards I don't think it worth of an effort.

**Comment by xkrt** [comment]
@dmitry-a-morozov thanks for the answer. It will be good idea to explicitly note in README that "Current Mono not supported".
Any plans to support .NET Core?

**Comment by dmitry-a-morozov** [comment]
@xkrt Yes, there are plans to support .NET Core. But I'll wait until F# compiler will get decent on .NET Core. 
Speaking of Mono, if they will improve their System.Data.SqlClient implementation I will certainly reconsider. They can do it by taking fully managed implementation that shipped with .NET Core.

**Comment by smoothdeveloper** [comment]
@xkrt if you have a chance, could you check if this issue is still relevant?
